live-event-app/live-event-app/RootViewController.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import PubNubChat
+import PubNubChatComponents
+
+class RootViewController: UIViewController {
+  
+  private let provider: PubNubChatProvider
+  private let channelId: String
+  
+  private var portraitConstraints: [NSLayoutConstraint] = []
+  private var landscapeConstraints: [NSLayoutConstraint] = []
+  private weak var scene: UIWindowScene?
+  
+  init(
+    channelId: String,
+    provider: PubNubChatProvider,
+    scene: UIWindowScene
+  ) {
+    self.channelId = channelId
+    self.provider = provider
+    self.scene = scene
+    
+    super.init(nibName: nil, bundle: nil)
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+    
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    guard let scene = scene else {
+      return
+    }
+    // Creates the default MessageList component
+    guard let messageListViewModel = try? provider.messageListComponentViewModel(pubnubChannelId: channelId) else {
+      preconditionFailure("Could not create intial view models")
+    }
+        
+    // Sets the custom title view for the MessageList component
+    messageListViewModel.customNavigationTitleString = nil
+    messageListViewModel.customNavigationTitleView = { [weak self] viewModel in
+      self?.getMessageListCustomTitleView() ?? UIView()
+    }
+    
+    // Creates MessageList and LiveStream view controllers
+    let messageListViewController = messageListViewModel.configuredComponentView()
+    let messageListContainerViewController = UINavigationController(rootViewController: messageListViewController)
+    let streamViewController = LiveStreamViewController()
+            
+    // Combine the content from MessageList and LiveStream view controllers using a Container View Controller
+    self.addChild(messageListContainerViewController)
+    self.addChild(streamViewController)
+    self.view.addSubview(messageListContainerViewController.view)
+    self.view.addSubview(streamViewController.view)
+            
+    messageListContainerViewController.didMove(toParent: self)
+    streamViewController.didMove(toParent: self)
+        
+    messageListContainerViewController.view.translatesAutoresizingMaskIntoConstraints = false
+    streamViewController.view.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
what do you think about making setup for streamViewController then for messageListContrainerViewController? In that way we can make separate functions and remove comments

Reply:
I'll provide the updated version of this PR shortly to review so you can check it out

Reply:
I removed a comment. Both `messageListContainerViewController` and `streamViewController` depend on each other. For instance, `messageListContainerViewController` top anchor is bounded to the bottom anchor of the `streamViewController` in portrait mode, so there's always a dependency between them. I checked your suggestion and the attempt to split applying constraints into two separate functions produced more code than it was originally.

Reply:
Thats true that they depends on each other but it is not obstacle to create one setup then another, when activating constraints only in one place we have dependency
`streamViewController.view.bottomAnchor.constraint(equalTo: messageListContainerViewController.view.topAnchor)`
and this line requires both views in the same views hierarchy. Sometimes more code is cleaner then too long function like this one.